### PR TITLE
Make HUP work on LUKS-encrypted systems

### PR DESCRIPTION
**Pull request by mtoman**
Tweak the device-specific bits to handle HUP on systems with FDE/LUKS

### CODE REVIEW

File: layers/meta-balena-genericx86/recipes-core/images/balena-image.bbappend [hunk 1]
```diff
@@ -2,10 +2,16 @@ include balena-image.inc
 
 BALENA_BOOT_PARTITION_FILES_append = " \
 	grub.cfg_internal:/EFI/BOOT/grub.cfg \
+	grub.cfg_internal_luks:/EFI/BOOT/grub-luks.cfg \
```
Comment (alexgg) [inline layers/meta-balena-genericx86/recipes-core/images/balena-image.bbappend:5]: These is not only for this device type I guess. Other DTs that use a TPM based luks partition encryption would also need it, so what about moving it to meta-balena?

```
BALENA_BOOT_PARTITION_FILES:append = "${@oe.utils.conditional('SIGN_API','','',' grub.cfg_internal_luks:/EFI/BOOT/grub-luks.cfg',d)}"
```

File: layers/meta-balena-genericx86/recipes-support/hostapp-update-hooks/files/999-resin-boot-cleaner [hunk 2]
```diff
@@ -28,15 +29,30 @@ if [ "$DURING_UPDATE" = "1" ]; then
...
+		type=$(lsblk -no TYPE "${boot_dev}")
+		if [ "${type}" = "crypt" ]; then
+			boot_part="$(lsblk -sJ "${boot_dev}" | jq -r '.blockdevices[].children[].name')"
+		else
+			boot_part="$(lsblk -sJ "${boot_dev}" | jq -r '.blockdevices[].name')"
+		fi
+		device=$(lsblk "/dev/$boot_part" -dnlo PKNAME)
+		pttype=$(get_part_table_type "${device}")
+
+		if [ "${pttype}" = "mbr" ]; then
```
Comment (alexgg) [inline layers/meta-balena-genericx86/recipes-support/hostapp-update-hooks/files/999-resin-boot-cleaner:42]: Why is this done only for MBR partitions? Even if this device type does not use GPT partitions we should consider it. I am wondering whether this hook that is used in several device types should be moved to meta-balena. What do you think?

File: layers/meta-balena-genericx86/recipes-support/resin-init/resin-init-board/resin-init-board [hunk 3]
```diff
@@ -2,10 +2,22 @@
...
+    boot_part=$(findmnt --noheadings --canonicalize --output SOURCE "/mnt/boot")
+    type=$(lsblk -no TYPE "${boot_part}")
+    if [ "${type}" = "crypt" ]; then
+        device="$(lsblk -sJ "${boot_part}" | jq -r '.blockdevices[].children[].name')"
+    else
+        device="$(lsblk -sJ "${boot_part}" | jq -r '.blockdevices[].name')"
+    fi
+    pttype=$(get_part_table_type "${device}")
+
+    if [ "$pttype" = "mbr" ]; then
```
Comment (alexgg) [inline layers/meta-balena-genericx86/recipes-support/resin-init/resin-init-board/resin-init-board:18]: Why checking for MBR only?

**Review by alexgg — CHANGES_REQUESTED** [review]

File: layers/meta-balena-genericx86/recipes-core/images/balena-image.bbappend (hunk 1 above)
Comment (mtoman) [reply layers/meta-balena-genericx86/recipes-core/images/balena-image.bbappend]: Moved to https://github.com/balena-os/meta-balena/pull/2880

File: layers/meta-balena-genericx86/recipes-core/images/balena-image.bbappend [hunk 4]
```diff
@@ -2,10 +2,16 @@ include balena-image.inc
...
 	grub.cfg_internal:/grub/grub.cfg \
 	"
 
 IMAGE_INSTALL_append_genericx86-64-ext =" \
+    peak                                  \
+    cryptsetup                            \
+    lvm2-udevrules                        \
+    tpm2-tools                            \
+    libtss2-tcti-device                   \
```
[the remark this answers is not in the document]
Comment (mtoman) [reply layers/meta-balena-genericx86/recipes-core/images/balena-image.bbappend]: Moved the TPM-related bits to https://github.com/balena-os/meta-balena/pull/2880

I am still not sure about the `peak` driver though. The sources are in meta-balena but I feel like it should be pulled in by the device types that need it, it is a 3rd party CAN driver, I do not think we necessarily want it everywhere.

File: layers/meta-balena-genericx86/recipes-support/resin-init/resin-init-board/resin-init-board (hunk 3 above)
Comment (mtoman) [reply layers/meta-balena-genericx86/recipes-support/resin-init/resin-init-board/resin-init-board:18]: Same as above, the `dd` under the if would corrupt a GPT partition.

File: layers/meta-balena-genericx86/recipes-core/images/balena-image.bbappend (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexgg) [reply layers/meta-balena-genericx86/recipes-core/images/balena-image.bbappend]: I agree @mtoman let's leave the `peak` driver device specific.

File: layers/meta-balena-genericx86/recipes-support/hostapp-update-hooks/files/999-resin-boot-cleaner (hunk 2 above)
Comment (alexgg) [reply layers/meta-balena-genericx86/recipes-support/hostapp-update-hooks/files/999-resin-boot-cleaner:42]: @mtoman as discussed, let's do `generic-amd64` as it's a better fit. I don't specially like having two different partition tables used in the same DT.

**Comment by mtoman** [comment]
We decided to switch to generic-amd64 for this feature

**Pull request closed without merging**